driver-core/src/test/unit/com/mongodb/internal/connection/SocketChannelStreamSpecification.groovy

```diff
@@ -0,0 +1,65 @@
+package com.mongodb.internal.connection
+
+import com.mongodb.MongoSocketOpenException
+import com.mongodb.ServerAddress
+import com.mongodb.connection.BufferProvider
+import com.mongodb.connection.SocketSettings
+import com.mongodb.connection.SslSettings
+import spock.lang.Specification
+
+class SocketChannelStreamSpecification extends Specification {
+    def 'should successfully connect with working ip address group'() {
```

Comment:
I wasn't sure how to combine this test with the one below it. There are a few similar variables in each. I do think that keeping them separate makes the test messages a bit more clear, and the tests are a bit more easy to read. But if you guys have any suggestions, I'm definitely open to anything!

Reply:
Also I just realized that I should probably move this test to the functional folder since it might be considered an integration test.

Reply:
@rozza think we should combine the two tests using a Spock `where` clause?

Reply:
I'd be tempted to merge the tests with a where clause. It would have to produce a closure that in the test you can pass the `serverAddress, socketSettings, sslSettings, socketFactory, bufferProvider`  values into.

Reply:
I'll defer this to @rozza